tests_oval_graph/test_arf_xml_parser/test_arf_xml_parser.py

```diff
@@ -9,6 +10,26 @@ def get_arf_report_path(src="global_test_data/ssg-fedora-ds-arf.xml"):
     return str(Path(__file__).parent.parent / src)
 
 
+def generate_arf_report_without_system_data():
+    arf_report_path = get_arf_report_path()
+    data = None
+    with open(arf_report_path, "r", encoding="utf-8") as arf_report:
+        data = arf_report.readlines()
+    tmp_patch = Path(tempfile.gettempdir()) / "arf_report_without_system_data.xml"
+    if tmp_patch.exists():
+        return str(tmp_patch)
+    with open(tmp_patch, "w", encoding="utf-8") as new_arf_report:
+        skip = False
+        for row in data:
+            if "<system_data>" in row:
+                skip = True
+            if not skip:
+                new_arf_report.write(row)
+            if "</system_data>" in row:
+                skip = False
+    return str(tmp_patch)
```

Comment:
I love that you have created a test. But I find this way of testing little crazy. You take a 12 MB ARF file, manually remove some lines from there, which effectively creates an invalid or nonsense file and then you test something on that. I guess people will have a hard time understanding this test. If I would write a test for it, I would create a very minimalist SCAP source data stream which would contain just a single simple rule, no additional bells and whistles, and then I would evaluate it so that the generated ARF wouldn't contain any collected object which would lead to missing system_data element and then I would put this file to the repository. The rule can be anything simple like requiring a non-existing file.